test/Generator/TypeGeneratorTest.php

```diff
@@ -239,88 +258,149 @@ public function invalidType()
             ['1foo'],
             ['foo\\1foo'],
             ['?foo\\bar|null'],
+            ['?foo\\bar&null'],
             ['*'],
             ["\0"],
             ['\\array'],
             ['\\Array'],
             ['\\ARRAY'],
             ['\\array|null'],
+            ['\\array&null'],
             ['null|\\array'],
+            ['null&\\array'],
             ['?array|null'],
+            ['?array&null'],
             ['\\callable'],
             ['\\Callable'],
             ['\\CALLABLE'],
             ['\\callable|null'],
+            ['\\callable&null'],
             ['null|\\callable'],
+            ['null&\\callable'],
             ['?callable|null'],
+            ['?callable&null'],
             ['\\string'],
             ['\\String'],
             ['\\STRING'],
             ['\\string|null'],
+            ['\\string&null'],
             ['null|\\string'],
+            ['null&\\string'],
             ['?string|null'],
+            ['?string&null'],
             ['\\int'],
             ['\\Int'],
             ['\\INT'],
             ['\\int|null'],
+            ['\\int&null'],
             ['null|\\int'],
+            ['null&\\int'],
```

Comment:
intersection types are not nullable, arent they?

Reply:
This is in `invalidType()` :+1: